examples/hiop/ex9p.cpp

```diff
@@ -0,0 +1,819 @@
+   {
+      char vishost[] = "localhost";
+      int  visport   = 19916;
+      sout.open(vishost, visport);
+      if (!sout)
+      {
+         if (myid == 0)
+            cout << "Unable to connect to GLVis server at "
+                 << vishost << ':' << visport << endl;
+         visualization = false;
+         if (myid == 0)
+         {
+            cout << "GLVis visualization disabled.\n";
+         }
+      }
+      else
+      {
+         sout << "parallel " << num_procs << " " << myid << "\n";
+         sout.precision(precision);
+         sout << "solution\n" << *pmesh << *u;
+         sout << "pause\n";
+         sout << flush;
+         if (myid == 0)
+            cout << "GLVis visualization paused."
+                 << " Press space (in the GLVis window) to resume it.\n";
+      }
+   }
+
+   Vector M_rowsums(m->Size());
+   m->SpMat().GetRowSums(M_rowsums);
+
+   // 10. Define the time-dependent evolution operator describing the ODE
+   //     right-hand side, and perform time-integration (looping over the time
+   //     iterations, ti, with a time-step dt).
+   FE_Evolution adv(*M, *K, *B, *k, M_rowsums);
+
+   double t = 0.0;
+   adv.SetTime(t);
+   ode_solver->Init(adv);
+
+   *u = *U;
+
+   // Compute initial volume.
+   const double vol0_loc = M_rowsums * (*u);
+   double vol0;
+   MPI_Allreduce(&vol0_loc, &vol0, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+
+   bool done = false;
+   for (int ti = 0; !done; )
+   {
+      double dt_real = min(dt, t_final - t);
+      adv.SetTimeStep(dt_real);
+      ode_solver->Step(*U, t, dt_real);
+      ti++;
+
+      done = (t >= t_final - 1e-8*dt);
+
+      if (done || ti % vis_steps == 0)
+      {
+         if (myid == 0)
+         {
+            cout << "time step: " << ti << ", time: " << t << endl;
+         }
+
+         // 11. Extract the parallel grid function corresponding to the finite
+         //     element approximation U (the local solution on each processor).
+         *u = *U;
+
+         if (visualization)
+         {
+            sout << "parallel " << num_procs << " " << myid << "\n";
+            sout << "solution\n" << *pmesh << *u << flush;
+         }
+
+         if (visit)
+         {
+            dc->SetCycle(ti);
+            dc->SetTime(t);
+            dc->Save();
+         }
+      }
+   }
+
+   // Print the error vs exact solution.
+   const double max_error = u->ComputeMaxError(u0),
+                l1_error  = u->ComputeL1Error(u0),
+                l2_error  = u->ComputeL2Error(u0);
+   if (myid == 0)
+   {
+      std::cout << "Linf error = " << max_error << endl
+                << "L1   error = " << l1_error << endl
+                << "L2   error = " << l2_error << endl;
+   }
+
+   // Print error in volume.
+   const double vol_loc = M_rowsums * (*u);
+   double vol;
+   MPI_Allreduce(&vol_loc, &vol, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+   if (myid == 0)
+   {
+      std::cout << "Vol  error = " << vol - vol0 << endl;
+   }
+
+   /*
```

Comment:
This block should be either removed, or un-comented, or made optional with a command-line argument.